src/waitress/channel.py

```diff
@@ -349,7 +349,7 @@ def write_soon(self, data):
                     raise ClientDisconnected
                 num_bytes = len(data)
 
-                if data.__class__ is ReadOnlyFileBasedBuffer:
+                if isinstance(data, ReadOnlyFileBasedBuffer):
```

Comment:
There was a reason this was a `data.__class__` check instead of `isinstance`, and something broke in the past when I tried to update this. I just can't remember what exactly.

Reply:
I wasn't able to find anything in the history. That line goes back well over a decade, and Chris didn't leave any explanation of why it uses `.__class__ is ...` rather than `isinstance(...)`. I can revert the code to use the former and instead leave a "there be dragons" comment here and in `task.py`, if that works?

Reply:
It was probably me avoiding a funccall for performance reasons while I was doing profiling.

Reply:
I'm a big fan of "if tests pass..." so let's let it ride for now.

I will re-review this tomorrow when I am not sleep deprived, and hope that I can remember what the issue was.